src/main.rs

```diff
@@ -103,55 +103,29 @@ fn main() -> crate::common::CliResult {
 
     let cli = match Cmd::try_parse() {
         Ok(cli) => cli,
-        Err(error) => {
-            match error.kind() {
-                clap::error::ErrorKind::DisplayHelp | clap::error::ErrorKind::DisplayVersion => {}
-                _ => match crate::js_command_match::JsCmd::try_parse() {
+        Err(error) => match error.kind() {
+            clap::error::ErrorKind::DisplayHelp | clap::error::ErrorKind::DisplayVersion => {
+                error.exit()
+            }
+            clap::error::ErrorKind::InvalidSubcommand => {
+                match crate::js_command_match::JsCmd::try_parse() {
                     Ok(js_cmd) => {
-                        match js_cmd.rust_command_generation() {
-                            Ok((vec_cmd, success_message)) => {
-                                eprintln!("{success_message}");
-                                eprintln!();
-                                eprintln!(
-                                    "    {}",
-                                    shell_words::join(
-                                        std::iter::once(near_cli_exec_path).chain(vec_cmd)
-                                    )
-                                    .yellow()
-                                );
-                            }
-                            Err(err) => {
-                                eprintln!("{}", err);
-                            }
-                        }
-                        std::process::exit(1);
+                        let vec_cmd = js_cmd.rust_command_generation();
+                        let cmd = &shell_words::join(
+                            std::iter::once(near_cli_exec_path.to_owned()).chain(vec_cmd),
+                        );
+                        Cmd::try_parse_from(cmd).unwrap()
                     }
                     Err(error) => {
-                        if let clap::error::ErrorKind::DisplayHelp = error.kind() {
-                            error.exit();
-                        }
-                        if let Some(cmd) = std::env::args().nth(1) {
-                            match cmd.as_str() {
```

Comment:
do not really understand what was happening here, but removing it does not seam to impact the tests or functionallity

Reply:
@gagdiez Please, bring it back. Without these, if you just enter a slightly invalid JS command you will get an unhelpful error message.

Reply:
I understood what was happening. I implemented a simpler logic that seams to handle errors correctly.

the logic now is:

```
if RS and JS cmd failed to parse the subcommand, we try an external sub-command execution

if JS successfully managed to parse the subcommand, then we let the `JSCmd` handle the error

otherwise we let `RS Cmd` handle the error
```

since there only 3 possibilities (both failed to parse the sub-command, JS failed, RS failed), then we have covered all cases.

I tested it manually and it seams to work fine